include/RAJA/pattern/forall.hpp

```diff
@@ -119,6 +127,97 @@ namespace RAJA
 /*!
  ******************************************************************************
  *
+ * \brief The RAJA::wrap layer unwraps dynamic policies before dispatch
+ *
+ ******************************************************************************
+ */
+namespace wrap {
+
+/*!
+ ******************************************************************************
+ *
+ * \brief Generic dispatch over containers with a value-based policy
+ *
+ ******************************************************************************
+ */
+template <typename EXEC_POLICY_T, typename Container, typename LOOP_BODY>
+RAJA_INLINE void forall(EXEC_POLICY_T&& p, Container&& c, LOOP_BODY loop_body)
+{
+#if defined(RAJA_ENABLE_CHAI)
+  chai::ArrayManager* rm = chai::ArrayManager::getInstance();
+  auto space = detail::get_space< typename std::remove_reference<decltype(p)>::type >::value;
+  rm->setExecutionSpace(space);
+#endif
+
+  typename std::remove_reference<decltype(loop_body)>::type body = loop_body;
```

Comment:
Definitely. In fact, we probably want to go through and ensure we are using `&&`  as much as possible elsewhere. We could even move this inside the `ifdef` so that a copy is only made when CHAI is used.